addons/pos_sale/models/pos_order.py

```diff
@@ -60,7 +60,13 @@ def sync_from_ui(self, orders):
                         self.env['sale.advance.payment.inv']._prepare_down_payment_section_values(sale_order_origin)
                     )
                 order_reference = line.name
-                sale_order_line_description = _("Down payment (ref: %(order_reference)s on \n %(date)s)", order_reference=order_reference, date=line.order_id.date_order.strftime('%m-%d-%y'))
+
+                lang = line.env.lang
+                if not order.partner_id.is_public and order.partner_id.lang and order.partner_id.lang != lang:
+                    lang = order.partner_id.lang
+                    line = line.with_context(lang=lang)
```

Comment:
was the original error string in the wrong language also? if yes, this makes sense (if no we could just use the lang of self.env.lang)

also is the is_public condition necessary? I didn't know you could do pos order with public user. but if they can and it makes sense to not use their lang, ok.

Reply:
> also is the is_public condition necessary? I didn't know you could do pos order with public user. but if they can and it makes sense to not use their lang, ok.

I have removed the condition for is_public as suggested.

> was the original error string in the wrong language also? if yes, this makes sense (if no we could just use the lang of self.env.lang)

No the original error was in the correct language, the system language (and not the customer language). However, the date was always formatted in `%m-%d-%y`, and now it's formatted according to the customer language.

Honestly not sure if the date should be formatted according to the customer's language, or to the system's language. But seeing from the sale_order_line.py logic, it's formatted according to the customer's language, so I kept the same logic.